Use TreeCellRenderers to prevent HTML injection issues

psiinon opened the pull request:
Fixes #3224

kingthorin commented on src/org/zaproxy/zap/view/SiteMapTreeCellRenderer.java:
```diff
@@ -71,23 +77,25 @@ public Component getTreeCellRendererComponent(JTree tree, Object value,
+			} else {
+				setPreferredSize(null);	// clears the prefered size, making the node visible
+				super.getTreeCellRendererComponent(tree, value, sel, expanded, leaf, row, hasFocus);
+			}
```
Are these white space changes necessary?

kingthorin replied:
Are these white space changes necessary?

kingthorin commented on src/org/zaproxy/zap/view/SiteMapTreeCellRenderer.java:
```diff
@@ -121,17 +129,54 @@ public Component getTreeCellRendererComponent(JTree tree, Object value,
 				}
-				setIcon(DisplayUtils.getScaledIcon(icon));
+				
+				//setIcon(DisplayUtils.getScaledIcon(icon));
```
This comment could be removed?

thc202 commented on src/org/parosproxy/paros/core/scanner/Alert.java:
```diff
@@ -688,6 +692,31 @@ public int getRisk() {
+    public ImageIcon getIcon() {
+        if (confidence == Alert.CONFIDENCE_FALSE_POSITIVE) {
+            // Special case - theres no risk - use the green flag
+            return DisplayUtils.getScaledIcon(new ImageIcon(Constant.OK_FLAG_IMAGE_URL));
```
It could use the same method as the other statements.

thc202 commented on src/org/zaproxy/zap/view/SiteMapTreeCellRenderer.java:
```diff
@@ -58,9 +62,11 @@
 
 	public SiteMapTreeCellRenderer(List<SiteMapListener> listeners) {
 		this.listeners = listeners;
+		this.component = new JPanel();
```
There's an extra spacing between the nodes, that might needs some tweaks (to look as before).

thc202 commented on src/org/zaproxy/zap/view/SiteMapTreeCellRenderer.java:
```diff
@@ -71,23 +77,25 @@ public Component getTreeCellRendererComponent(JTree tree, Object value,
 			boolean sel, boolean expanded, boolean leaf, int row,
 			boolean hasFocus) {
 
+		component = new JPanel();
```
It could call `component.removeAll()` instead.

thc202 commented on src/org/zaproxy/zap/view/SiteMapTreeCellRenderer.java:
```diff
@@ -71,23 +77,25 @@ public Component getTreeCellRendererComponent(JTree tree, Object value,
 			boolean hasFocus) {
 
+		component = new JPanel();
+		component.setOpaque(false);
```
This can be called once (in the constructor).

thc202 commented on src/org/zaproxy/zap/view/SiteMapTreeCellRenderer.java:
```diff
@@ -71,23 +77,25 @@ public Component getTreeCellRendererComponent(JTree tree, Object value,
 			// folder / file icons with scope 'target' if relevant
 			if (node.isRoot()) {
-				setIcon(DisplayUtils.getScaledIcon(ROOT_ICON));	// 'World' icon
+				component.add(new JLabel(DisplayUtils.getScaledIcon(DisplayUtils.getScaledIcon(ROOT_ICON)))); // 'World' icon
```
Could call warp here too?

psiinon commented:
Hopefully addressed all of the comments. The Sites tree spacing isnt exactly the same as before (its actually slightly closer together) but I think it looks ok - wdyt?

thc202 commented:
Sites tree spacing looks good to me.
